packages/odyssey-react-mui/src/labs/SideNav/SideNav.tsx

```diff
@@ -333,62 +381,43 @@ const SideNav = ({
       sideNavItems.map((item) => ({
         ...item,
         children: item.children?.map((childProps) => (
-          <SideNavItemContent
+          <SideNavItemContentContext.Provider
+            value={{ isCompact: isCompact }}
             key={childProps.id}
-            scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
-            {...childProps}
-          />
+          >
+            <SideNavItemContent
+              key={childProps.id}
+              scrollRef={getRefIfThisIsFirstNodeWithIsSelected(childProps.id)}
+              {...childProps}
+            />
+          </SideNavItemContentContext.Provider>
         )),
       })),
-    [getRefIfThisIsFirstNodeWithIsSelected, sideNavItems],
+    [getRefIfThisIsFirstNodeWithIsSelected, sideNavItems, isCompact],
   );
 
-  const sideNavCollapseHandler = useCallback(() => {
-    setSideNavCollapsed(!isSideNavCollapsed);
-    onCollapse?.();
-  }, [isSideNavCollapsed, setSideNavCollapsed, onCollapse]);
-
   const sideNavExpandClickHandler = useCallback(() => {
+    isSideNavCollapsed ? onExpand?.() : onCollapse?.();
     setSideNavCollapsed(!isSideNavCollapsed);
-    onExpand?.();
-  }, [isSideNavCollapsed, setSideNavCollapsed, onExpand]);
+  }, [isSideNavCollapsed, setSideNavCollapsed, onExpand, onCollapse]);
 
   const sideNavExpandKeyHandler = useCallback(
     (event: KeyboardEvent<HTMLDivElement>) => {
       if (event?.key === "Enter" || event?.code === "Space") {
         event.preventDefault();
+        isSideNavCollapsed ? onExpand?.() : onCollapse?.();
         setSideNavCollapsed(!isSideNavCollapsed);
-        onExpand?.();
       }
     },
-    [isSideNavCollapsed, setSideNavCollapsed, onExpand],
-  );
-
-  const expandLeftIconStyles = useMemo(
-    () => ({
-      fontSize: "1em",
-      margin: `0 ${odysseyDesignTokens.Spacing1}`,
-    }),
-    [odysseyDesignTokens],
+    [isSideNavCollapsed, setSideNavCollapsed, onExpand, onCollapse],
   );
 
   return (
-    <SideNavContainer expandedWidth={expandedWidth}>
-      <SideNavCollapsedContainer
-        tabIndex={0}
-        role="button"
-        odysseyDesignTokens={odysseyDesignTokens}
-        isSideNavCollapsed={isSideNavCollapsed}
-        onClick={sideNavExpandClickHandler}
-        onKeyDown={sideNavExpandKeyHandler}
-        data-se="collapsed-region"
-        data-aria-label="expand side navigation"
-      >
-        <ExpandLeftIcon sx={expandLeftIconStyles} />
-      </SideNavCollapsedContainer>
+    <SideNavContainer>
       <SideNavExpandContainer
         odysseyDesignTokens={odysseyDesignTokens}
         isSideNavCollapsed={isSideNavCollapsed}
+        tabIndex={-1}
```

Comment:
Just to understand, why are we removing this item from tabbing?

Reply:
removed. Added it for some testing.